sdk/core/core/inc/az_json.h

```diff
@@ -182,81 +196,233 @@ AZ_NODISCARD az_result az_json_token_get_string(az_json_token const* token, az_s
+                               .need_comma = false,
+                               .token_kind = AZ_JSON_TOKEN_NONE,
+                               .bit_stack = { 0 },
+                               .options
+                               = options == NULL ? az_json_builder_options_default() : *options,
+                           } };
   return AZ_OK;
 }
 
-/*
- * @brief az_json_builder_span_get returns the az_span containing the final JSON object.
+/**
+ * @brief Returns the #az_span containing the JSON text written to the underlying buffer so far.
+ *
+ * @param[in] json_builder A pointer to an #az_json_builder instance wrapping the destination
+ * buffer.
+ *
+ * @note Do NOT modify or override the contents of the returned #az_span unless you are no longer
+ * building JSON text into it.
  *
- * @param json_builder A pointer to an az_json_builder instance wrapping the JSON buffer.
- * @return an az_span containing the final JSON object.
+ * @return An #az_span containing the JSON text built so far.
  */
-AZ_NODISCARD AZ_INLINE az_span az_json_builder_span_get(az_json_builder const* json_builder)
+AZ_NODISCARD AZ_INLINE az_span az_json_builder_get_json(az_json_builder const* json_builder)
 {
-  return az_span_slice(json_builder->_internal.json, 0, json_builder->_internal.length);
+  // TODO: Consider precondition validation of the state before returning.
+  return az_span_slice(
+      json_builder->_internal.destination_buffer, 0, json_builder->_internal.bytes_written);
 }
 
-/*
- * @brief az_json_builder_append_token appends an az_json_token to the JSON buffer.
+/**
+ * @brief Appends the UTF-8 text value (as a JSON string) into the buffer.
  *
- * @param json_builder A pointer to an az_json_builder instance containing the buffer to append the
- * token to.
- * @param token A pointer to the az_json_token to append.
- * @return AZ_OK if the token was appended successfully.<br>
- * AZ_ERROR_BUFFER_OVERFLOW if the buffer is too small.
+ * @param[in] json_builder A pointer to an #az_json_builder instance containing the buffer to append
+ * the string value to.
+ * @param[in] value The UTF-8 encoded value to be written as a JSON string. The value is escaped
+ * before writing.
+ *
+ * @remarks If \p value is #AZ_SPAN_NULL, the JSON `null` value is written, as if the
+ * #az_json_builder_append_null() method was called.
+ *
+ * @return An #az_result value indicating the result of the operation:
+ *         - #AZ_OK if the string value was appended successfully
+ *         - #AZ_ERROR_INSUFFICIENT_SPAN_SIZE if the buffer is too small
  */
-AZ_NODISCARD az_result
-az_json_builder_append_token(az_json_builder* json_builder, az_json_token token);
+AZ_NODISCARD az_result az_json_builder_append_string(az_json_builder* json_builder, az_span value);
 
-/*
- * @brief az_json_builder_append_object appends a JSON to the JSON buffer.
+// TODO: Consider adding char* overloads to make passing string literals as property names easier.
```

Comment:
I lean on the side of NOT doing this. We have the az_span_from_str function and the AZ_SPAN_FROM_STRING macros for this and we want to encourage the use of spans as they are faster/smaller (no calls to strlen).

Reply:
Sounds good. The only reason I brought this up is because it is super common for folks to append **string literals** (especially for property names) and the code seemed a bit bloated for that particular case (as a usability concern). In comparison, passing a literal directly is kind of nice.

```C
az_result res = az_json_builder_append_property_name(&builder, AZ_SPAN_FROM_STR("int-max"));
// vs.
az_result res = az_json_builder_append_property_name(&builder, "int-max");
```